### PR TITLE
auto-remove duplicate chapters

### PR DESCRIPTION
**Pull request by AriaMoradi**

### CODE REVIEW

**Comment by Syer10** [comment]
I took a gander at this, I think it looks much cleaner
```kotlin
        val dbChapterCount = transaction { ChapterTable.select { ChapterTable.manga eq mangaId }.count() }
        if (dbChapterCount > chapterCount) { // we got some clean up due
            val dbChapterList = transaction { ChapterTable.select { ChapterTable.manga eq mangaId }.toList() }
            val chapterUrls = chapterList.map { it.url }.toSet()

            dbChapterList.forEach { dbChapter ->
                if (
                    !chapterUrls.contains(dbChapter[ChapterTable.url]) || // is orphaned
                    dbChapterList.count { it[ChapterTable.url] == dbChapter[ChapterTable.url] } > 1 // is duplicate
                ) {
                    transaction {
                        PageTable.deleteWhere { PageTable.chapter eq dbChapter[ChapterTable.id] }
                        ChapterTable.deleteWhere { ChapterTable.id eq dbChapter[ChapterTable.id] }
                    }
                }
            }
        }
```
Though it may have issues where yours doesnt, not too sure if they are equal

**Comment by AriaMoradi** [comment]
your's is not deterministic, mine always makes sure there's a non-duplicate with bigger index exists to check with the next?

File: server/src/main/kotlin/suwayomi/tachidesk/manga/impl/Chapter.kt [hunk 1]
```diff
@@ -101,14 +102,21 @@ object Chapter {
...
-            val dbChapterList = transaction { ChapterTable.select { ChapterTable.manga eq mangaId }.toList() }
+            val dbChapterList = transaction {
+                ChapterTable.select { ChapterTable.manga eq mangaId }.orderBy(ChapterTable.url to ASC).toList()
+            }
             val chapterUrls = chapterList.map { it.url }.toSet()
 
-            dbChapterList.forEach { dbChapter ->
-                if (!chapterUrls.contains(dbChapter[ChapterTable.url])) {
+            for (i in dbChapterList.indices) {
+                val dbChapter = dbChapterList[i]
```
Comment (Syer10) [inline server/src/main/kotlin/suwayomi/tachidesk/manga/impl/Chapter.kt]:
```suggestion
```

File: server/src/main/kotlin/suwayomi/tachidesk/manga/impl/Chapter.kt [hunk 2]
```diff
@@ -101,14 +102,21 @@ object Chapter {
...
         if (dbChapterCount > chapterCount) { // we got some clean up due
-            val dbChapterList = transaction { ChapterTable.select { ChapterTable.manga eq mangaId }.toList() }
+            val dbChapterList = transaction {
+                ChapterTable.select { ChapterTable.manga eq mangaId }.orderBy(ChapterTable.url to ASC).toList()
+            }
             val chapterUrls = chapterList.map { it.url }.toSet()
 
-            dbChapterList.forEach { dbChapter ->
-                if (!chapterUrls.contains(dbChapter[ChapterTable.url])) {
+            for (i in dbChapterList.indices) {
```
Comment (Syer10) [inline server/src/main/kotlin/suwayomi/tachidesk/manga/impl/Chapter.kt]:
```suggestion
            dbChapterList.forEachIndexed { index, dbChapter ->
```

File: server/src/main/kotlin/suwayomi/tachidesk/manga/impl/Chapter.kt [hunk 3]
```diff
@@ -101,14 +102,21 @@ object Chapter {
...
+                val dbChapter = dbChapterList[i]
+
+                if (
+                    !chapterUrls.contains(dbChapter[ChapterTable.url]) || // is orphaned
+                    (i + 1 < dbChapterList.size && dbChapter[ChapterTable.url] == dbChapterList[i + 1][ChapterTable.url]) // is duplicate
```
Comment (Syer10) [inline server/src/main/kotlin/suwayomi/tachidesk/manga/impl/Chapter.kt]:
```suggestion
                    (index < dbChapterList.lastIndex && dbChapter[ChapterTable.url] == dbChapterList[index + 1][ChapterTable.url]) // is duplicate
```

**Review by Syer10 — CHANGES_REQUESTED** [review]
I dont like non-descriptive variables like i, changed to index. Also cleaned up the code

**Review by Syer10 — APPROVED** [review]

**Pull request merged**